src/reanimated2/threads.ts

```diff
@@ -164,13 +164,10 @@ export function runOnJS<A extends any[], R>(
     // reference to the original remote function in the `__remoteFunction` property.
     fun = fun.__remoteFunction;
   }
-  if (!_WORKLET) {
-    return fun;
-  }
   return (...args) => {
     _scheduleOnJS(
       fun,
-      args.length > 0 ? makeShareableCloneOnUIRecursive(args) : undefined
+      args.length > 0 ? makeShareableCloneOnUIRecursive(args) : []
```

Comment:
Instead of passing an empty array here, let's update typing of `_scheduleOnJS` function because `undefined` here means a fast path for remote functions without arguments:

https://github.com/software-mansion/react-native-reanimated/blob/d3b2b84df0d34338a5046f69a36dea7bf38480b7/Common/cpp/NativeModules/NativeReanimatedModule.cpp#L78-L87

Reply:
@tomekzaw Please check if I understood correctly.

Reply:
Yup, it's correct. Too bad `...args` causes a TypeError when `args` is `undefined`. We could also go with `...(args ?? [])` but I like your approach better.